src/lib/CheckType.hs

```diff
@@ -900,7 +905,8 @@ checkTabApp tabTy xs = go tabTy $ toList xs
     go :: Typer m => Type o -> [Atom i] -> m i o (Type o)
     go ty [] = return ty
     go ty (i:rest) = do
-      TabTy (b:>ixTy) resultTy <- return ty
+      TabTy (b :> IxType ixTy _) resultTy <- return ty
+      -- TODO: check type of Ix dict too
```

Comment:
Please add this?

Reply:
Actually, on second thought I think it's not needed. The type is supplied in the `o` space, meaning that it has already been checked. I'll delete the comment instead.